### PR TITLE
Debug/map features

### PR DESCRIPTION
**Pull request by GiftLanga**
separated map component.
used new map component on home screen and fullscreen.
added more button on the map component

### CODE REVIEW

File: waterlog/src/Components/Map.js [hunk 1]
```diff
@@ -150,17 +108,23 @@ class MapComponent extends Component {
...
               src={require("../images/recentre_icon_blue.png")}
               alt="re-center Map"
               onClick={() => {
-                const map = this.refs.map.leafletElement;
-                map.setView(mapOptions.centerPosition, mapOptions.defaultZoom)
+                this.refs.map.reCenter()
+              }} />
+            <img
+              className="icon-home"
+              src={require("../images/more_map_icon.png")}
```
Comment (FredJEhlers) [inline waterlog/src/Components/Map.js]: Please extract the images to a variable

Comment (FredJEhlers) [reply waterlog/src/Components/Map.js]: It works something like 
`import moreMapIcon from '../images/more_map_icon.png'`
then you can say 
src={moreMapIcon)

File: waterlog/src/Components/MapComponent.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+class MapComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.reCenter = this.reCenter.bind(this);
+    this.ref = this.refs.map;
+    this.weight = 2;
+    this.sign = 1;
+  }
+  reCenter() {
```
Comment (FredJEhlers) [inline waterlog/src/Components/MapComponent.js:49]: space between functios

File: waterlog/src/Components/MapComponent.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                          max={mapOptions.maxIntensity} />
+                      </div>);
+                  }
+                })()}
+              </div>
+              );
+          })()}
+          {(() => {
+            let icons = [];
+            console.log("g", this.props.mapsData)
```
Comment (FredJEhlers) [inline waterlog/src/Components/MapComponent.js]: logs

File: waterlog/src/Components/MapFullScreen.js [hunk 4]
```diff
@@ -107,78 +89,50 @@ class MapFullScreenComponent extends Component {
             </button>
           </div>
           <div className={`map-icon-div-layer2-fullscreen map-button-tab ${this.state.simpleView ? "" : "invisible"}`}>
-            
+
             <img
-              className="icon"
+              className="icon-home"
+              src={require("../images/more_map_icon.png")}
```
Comment (FredJEhlers) [inline waterlog/src/Components/MapFullScreen.js]: note previous comment on icons and variables

File: waterlog/src/Stylesheets/_map.scss [hunk 5]
```diff
@@ -11,9 +11,9 @@
 	flex: 1;
 	margin: 5px;
 	background-color: $map-btn;
-	height: 26px;
-	width: 90px;
-	border: none;
+	height: 30px;
+  width: 90px;
```
Comment (FredJEhlers) [inline waterlog/src/Stylesheets/_map.scss]: inconsistent indentation

Comment (FredJEhlers) [reply waterlog/src/Stylesheets/_map.scss]: through out the file

File: waterlog/src/Stylesheets/_map.scss [hunk 6]
```diff
@@ -116,13 +119,14 @@
...
 .icon {
-	width: 30px;
-	height: 30px;
-	margin: 2px 2px;
+  width: 35px;
+  height: 35x;
+  margin: 2px 2px;
 }
 
 .icon-home {
```
Comment (FredJEhlers) [inline waterlog/src/Stylesheets/_map.scss]: `icon` and `icon-home` is the same class

File: waterlog/src/icons/MapIcons.js [hunk 7]
```diff
@@ -30,12 +30,12 @@ export const sensorFaultIcon = new L.icon({
   iconSize: new L.Point(25, 25)
 });
 
-export const sensorRedIcon = new L.icon({
-  iconUrl: require('../images/Red_sensor_dark.gif'),
-  iconAnchor: null,
+export const selectedComponentIcon = new L.icon({
+  iconUrl: require('../images/selectedComponent.png'),
```
Comment (FredJEhlers) [inline waterlog/src/icons/MapIcons.js]: apply the same logic of icon request to this image

File: waterlog/src/utils.js [hunk 8]
```diff
@@ -1,186 +1,197 @@
...
-	northEast: northEast,
-	maxIntensity: 5,
-	centerPosition: [ -25.783425, 28.336046 ],
-	defaultZoom: 17,
-	heatBackgroundConst: heatBackgroundConst,
-	rectangleBounds: [ southWest.map((e) => e / heatBackgroundConst), northEast.map((e) => e * heatBackgroundConst) ]
+  southWest: southWest,
+  northEast: northEast,
+  maxIntensity: 5,
+  centerPosition: [-25.783425, 28.336046],
```
Comment (FredJEhlers) [inline waterlog/src/utils.js]: mimic what was done with `southWest` and `northEast `

**Review by FredJEhlers — CHANGES_REQUESTED** [review]

File: waterlog/src/Components/Map.js (hunk 1 above)
Comment (GiftLanga) [reply waterlog/src/Components/Map.js]: imported images into variable

File: waterlog/src/Components/MapComponent.js (hunk 2 above)
Comment (GiftLanga) [reply waterlog/src/Components/MapComponent.js:49]: added space

File: waterlog/src/Components/MapComponent.js (hunk 3 above)
Comment (GiftLanga) [reply waterlog/src/Components/MapComponent.js]: removed logs

File: waterlog/src/Components/MapFullScreen.js (hunk 4 above)
Comment (GiftLanga) [reply waterlog/src/Components/MapFullScreen.js]: implemented

File: waterlog/src/Stylesheets/_map.scss (hunk 5 above)
Comment (GiftLanga) [reply waterlog/src/Stylesheets/_map.scss]: indented

File: waterlog/src/Stylesheets/_map.scss (hunk 6 above)
Comment (GiftLanga) [reply waterlog/src/Stylesheets/_map.scss]: deleted icon-home

File: waterlog/src/icons/MapIcons.js (hunk 7 above)
Comment (GiftLanga) [reply waterlog/src/icons/MapIcons.js]: done

File: waterlog/src/utils.js (hunk 8 above)
Comment (GiftLanga) [reply waterlog/src/utils.js]: done

**Review by FredJEhlers — APPROVED** [review]

**Pull request merged**